src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -90,6 +92,28 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             TrackBulkFileOperations();
         }
 
+        private async Task InitializeWorkspaceFailureOutputWindowAsync()
+        {
+            var threadingContext = _componentModel.GetService<IThreadingContext>();
+            var asyncListenerProvider = _componentModel.GetService<IAsynchronousOperationListenerProvider>();
+            var asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.WorkspaceFailureLogger);
+
+            var brokeredServiceContainer = await this.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
```

Comment:
Does this need a SwitchToMainThreadAsync() prior to the GetServiceAsync?

Reply:
I recall that wasn't necessary, and I didn't encounter any issues with it, but I'll defer to @AArnott for confirmation.

Reply:
`GetServiceAsync` will switch to the UI thread when necessary, and once you get the object back, that's a free-threaded object. So no explicit switch is necessary here.